src/main/webapp/create-cache/create-cache.controller.js

```diff
@@ -0,0 +1,196 @@
+              var cmpNode = cache.data.compatibility.COMPATIBILITY;
+              cacheCreateController.createCompatibilityChildNode(compAddress, cmpNode.enabled, cmpNode.marshaller);
+            })
+            .then(function () {
+              var txAddress = angular.copy(address);
+              txAddress.push('transaction', 'TRANSACTION');
+              var txNode = cache.data.transaction.TRANSACTION;
+              cacheCreateController.createTransactionChildNode(txAddress, txNode.mode, txNode.locking, txNode['stop-timeout']);
+            })
+            .then(function () {
+              var fileStoreAddress = angular.copy(address);
+              fileStoreAddress.push('file-store', 'FILE_STORE');
+              var node = cache.data['file-store'].FILE_STORE;
+              cacheCreateController.createFileStoreChildNode(fileStoreAddress, node.shared, node.preload,
+                node.passivation, node['fetch-state'], node.purge, node['read-only'], node.singleton, node['max-entries'],
+                node['relative-to'], node.path);
+            })
+            .then(function () {
+              var jdbcStoreAddress = angular.copy(address);
+              jdbcStoreAddress.push('string-keyed-jdbc-store', 'STRING_KEYED_JDBC_STORE');
+              var node = cache.data['string-keyed-jdbc-store'].STRING_KEYED_JDBC_STORE;
+              cacheCreateController.createStringKeyedJDBCStoreChildNode(jdbcStoreAddress, node.shared, node.preload,
+                node.passivation, node['fetch-state'], node.purge, node['read-only'], node.singleton, node.datasource,
+                node.dialect, node['string-keyed-table']);
+            })
+            .then(function () {
+              //now that cache is created we have to refresh the cluster (new cache will be added to view)
+              $scope.currentCluster.refresh();
+              $state.go('clusterView', {clusterName: $scope.currentCluster.name});
+            })
+        });
+      };
+
+      $scope.createCache = function (){
+          var address = ['profile', 'clustered', 'subsystem', 'datagrid-infinispan', 'cache-container', $scope.currentCluster.name];
+          address.push($scope.newlyCreatedCache.type);
+          address.push($scope.newlyCreatedCache.name);
+          if ($scope.newlyCreatedCache.type === 'distributed-cache' || $scope.newlyCreatedCache.type === 'replicated-cache') {
+            $scope.createReplicatedOrDistributedCache($scope.newlyCreatedCache, address, $scope.newlyCreatedCache.type);
+          } else if ($scope.newlyCreatedCache.type === 'invalidation-cache' || $scope.newlyCreatedCache.type === 'local-cache') {
+            $scope.createInvalidationOrLocalCache($scope.newlyCreatedCache, address, $scope.newlyCreatedCache.type);
+          }
+      };
+
+      $scope.$on('createEnabled', function(e, data) {
+        $scope.isCreateEnabled = data;
+      });
+
+      $scope.$on('createCacheTypeSelected', function(e, data) {
```

Comment:
Same as above, why broadcasting? I would prefer passing attributes to directives if possible.